### ISSUE TITLE
boon-select-content?

### ISSUE THREAD

**Issue by subsetpark** [jyp/boon#101]
I'm trying to piece together a full understanding of all the different boon functionality, and the region selectors are the most mysterious to me.

One that's not touched on at all in the tutorial is `boon-select-content`, and the docstring for that function is a bit hard to parse.

Since it shows up in the keymap PDFs, I assume it's designed for general consumption.

Could you say a few words about its behaviour, and maybe give an example of how it's used?

**Comment by jyp** [comment]
boon-select-content is perhaps the least useful region selector. I barely ever use it. So let's talk about boon-select-borders first, which is explained in the tutorial (it is bound to 'a' below, but call boon-tutorial to see it adapted to your keybindings)
```
* ENCLOSURES (PARENS, ETC.)
---------------------------

Boon provides help with manipulating enclosures.

The command ’M-x boon-enclose’ (mnemonic: around) adds enclosures around a region.

It takes two arguments:

1. The kind of enclosure to use (’p’ for regular parentheses)

2. The region to surround.  This region is specified by the same
   language as the arguments to ’M-x boon-take-region’ or ’M-x boon-drop-mark’ commands.

>> With the cursor over a word, type ’M-x boon-enclose p s’ to enclose it in parens.

It is often useful to put enclosures around what has just been pasted.
The just pasted region is accessible with: ’'’ (single quote).

>> Delete a word with ’t f’, then paste it with ’s’.

>> Put parens around the just pasted word with: ’a p '’.

>> Inspect and customize the kind of enclosures available with:
   ’M-x customize-variable boon-enclosures’ ’<Return>’

It is also possible to remove enclosures with the ’a’ operator, which
transforms a region argument to its enclosure (the first and last characters).

Just delete the parentheses:

>> Move the cursor inside this (sexp).

>> Type ’t a x’.

Or delete the whole expression:

>> Move the cursor inside this (sexp).

>> Type ’t x’.

The ’a’ operator is especially useful when applied to ’x’, ’n’
and ’o’ --- but it works on any region.
```

boon-select-content is the dual of boon-select-borders: it transforms a region to its 'insides', by removing the 'borders'. Again, I find it next to useless.

**Comment by subsetpark** [comment]
@jyp thank you for that clarification! From my understanding, I think I can see why it would be included, as it's the dual of `select-borders`, and also why it wouldn't be very useful :)

Just to test my understanding: it seems to me that it would be much *more* useful if we didn't already have `select-inside_pairs`, is that right? 

Given a line like

`foo(bar)`

I can remove `bar` by `t z x` - but I can also remove `bar` by `t c`.

**Comment by evandromurilo** [comment]
@jyp Hello! Thanks for the package, I'm having great fun with it.

I can't figure out how to do something that I think is related to this issue. Given I have the following text (^ ) is where the cursor is located): `class="btn ^btn-sm"`

How would I go around replacing the text inside `""`? In VIM, for instance, I would type: `ci" text`.

**Comment by jyp** [comment]
@evandromurilo I had implemented the "quotes" selector in terms of the extend-region package, but did not bind it.
I just pushed the commit which binds it to `"`.  (3dd9669d01e48860bb7ea12be8c764a4bc6808ea) You can try `rz"` using the colemak bindings. With qwerty it is `sz"` I think.

**Comment by evandromurilo** [comment]
@jyp Worked great. Thanks.
